File: Algorithms/Android/app/src/main/java/com/example/androidalgorithms/testers/AlgorithmTester.java
package com.example.androidalgorithms.testers;

import android.util.Log;

public abstract class AlgorithmTester {
    public class Result {
        public int testSize;
        public String description;
        public double time;
    }

    public abstract String testDescription(int testSize);
    public abstract double test(int testSize);

    public Result[] testAll(int fromTestSize, int toTestSize) {
        Result[] res = new Result[toTestSize - fromTestSize + 1];
        for(int i = 0; i <= toTestSize - fromTestSize; i++) {
            int size = fromTestSize + i;
            res[i] = new Result();
            res[i].time = test(size);
            res[i].description = testDescription(size);
            res[i].testSize = size;

            Log.i("custom", String.format("Test %d (%d / %d) completed", size, i+1, res.length));
        }
        return res;
    }
}

File: Algorithms/flutter_algorithms/lib/testers/algorithm_tester.dart
class Result {
  final int testSize;
  final String description;
  final int time;

  const Result(this.testSize, this.description, this.time);
}

abstract class AlgorithmTester {
  String testDescription(int testSize);
  int test(int testSize);

  List<Result> testAll(int fromTestSize, int toTestSize) {
    List<Result> res = new List<Result>(toTestSize - fromTestSize + 1);
    for(int i = 0; i <= toTestSize - fromTestSize; i++) {
        int size = fromTestSize + i;
        res[i] = new Result(
          size,
          testDescription(size),
          test(size),
        );

        print("Test $size (${i+1} / ${res.length}) completed");
    }
    return res;
  }
}